### ISSUE TITLE
Validation is broken

### ISSUE THREAD

**Issue by RayHughes** [mailgun/mailgun-php#689]
When trying to utilize validation (parse, validate), an json decode error is thrown. 
```
Error (4) when trying to json_decode response"
```

**Comment by DavidGarciaCat** [comment]
Hi @RayHughes 

Do you mind providing a full copy of the code you're providing, replacing your key/domain, so we can see if there's any problem with the call?

Cheers,

**Comment by Mhitra** [comment]
Same issue here. 

$mgClient = \Mailgun\Mailgun::create(KEY);
$check = $mgClient->emailValidation()->validate('test <test@test.com>');

**Comment by csrui** [comment]
Sorry for hopping on @RayHughes train but I'm having the same issue so I'll provide an example.

example 1
```php
$mgClient = Mailgun::create( MAILGUN_APIKEY, 'https://api.eu.mailgun.net' );
$result  = $mgClient->emailValidation()->validate( $email, true );
```

example 2
```php
$configurator = new \Mailgun\HttpClient\HttpClientConfigurator();
$configurator->setApiKey( MAILGUN_APIKEY );
$configurator->setEndpoint( 'https://api.eu.mailgun.net' );

$mgClient = new Mailgun( $configurator, new \Mailgun\Hydrator\ArrayHydrator() );
$result   = $mgClient->emailValidation()->validate( $email, true );
```

In both cases, I tried with default endpoint and with false on the validate second parameter. 

The result is the API responding with a 200 status code but with a body that is just a string "Mailgun Magnificent API" which at this point I'm doubting a little bit :p

So what say you @DavidGarciaCat? Thank you for your help.

**Comment by DavidGarciaCat** [comment]
Hello, @csrui 

## TL;DR

**This is an open gap that still needs an update.**

Also, it's a duplicated issue, so closing this one now.

## In-depth details

Thank you very much for your post. I appreciate your example, trying to clarify the problem that didn't include a single line to make it easier to understand.

This is a duplicated case that I will close now. We can follow this case on the original thread here https://github.com/mailgun/mailgun-php/issues/619

Mailgun support team confirmed that it is not possible to get an email validation based on the current implementation because:
- Firstly, the API v3 is no longer working in the way that we developed the email validation process, and it needs to be upgraded to the API v4.
- Secondly, it's unfortunate but the API v4 implementation also requires either a unique implementation for this update or a major release upgrading as many endpoints as possible to keep consistency with the versions (that brings another layer of complexity due to it requires an update on a large volume of files).
- Lastly, the EU servers don't provide the email validation functionality, meaning that we need to instantiate two different versions of the client if we use EU servers for sendings.

Some time ago, I started an update to fix this bug. However, because of these restrictions that the support team confirmed later (when I was getting errors even after reviewing the logic), it was not possible to move forward easily.

If you want to contribute by sending a PR that follows the same approach (or at least a similar one, to keep consistency) that we have for the SDK, you are more than welcome, and I am sure that the community will appreciate it. My only comment is, as I said, try to keep consistency, due to it seems wrong making major changes to the whole SDK just to fix this functionality. So, we should prevent any changes that could cause a backward compatibility issue with the current implementations of the SDK v3.x.

Unfortunately, I can't invest too much time on active development, but I am happy to allocate some time on my agenda for PR reviews and tag new releases.

Once more, thanks for your time providing the example.

Cheers,

**Comment by manish-dholpuriya** [comment]
Mailgun\Exception\HydrationException Object
(
    [message:protected] => Error (4) when trying to json_decode response
    [string:Exception:private] => 
    [code:protected] => 0
    [file:protected] => /var/www/html/b.solution/mail-template/vendor/mailgun/mailgun-php/src/Hydrator/ModelHydrator.php
    [line:protected] => 40
    [trace:Exception:private] => Array
        (
            [0] => Array
                (
                    [file] => /var/www/html/b.solution/mail-template/vendor/mailgun/mailgun-php/src/Api/HttpApi.php
                    [line] => 76
                    [function] => hydrate
                    [class] => Mailgun\Hydrator\ModelHydrator
                    [type] => ->
                )

            [1] => Array
                (
                    [file] => /var/www/html/b.solution/mail-template/vendor/mailgun/mailgun-php/src/Api/Message.php
                    [line] => 63
                    [function] => hydrateResponse
                    [class] => Mailgun\Api\HttpApi
                    [type] => ->
                )

            [2] => Array
                (
                    [file] => /var/www/html/b.solution/mail-template/sendMail.php
                    [line] => 28
                    [function] => send
                    [class] => Mailgun\Api\Message
                    [type] => ->
                )

        )

    [previous:Exception:private] => 
    [xdebug_message] => ( ! ) Mailgun\Exception\HydrationException: Error (4) when trying to json_decode response in /var/www/html/b.solution/mail-template/vendor/mailgun/mailgun-php/src/Hydrator/ModelHydrator.php on line 40
Call Stack
#TimeMemoryFunctionLocation
10.0002361480{main}(  ).../sendMail.php:0
